src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentService.cs

```diff
@@ -134,7 +133,9 @@ public async Task UpdateContent(PublishContext context, params Guid[] releaseIds
 
             foreach (var publication in publications)
             {
-                await CachePublication(publication.Id, context, releaseIds);
+                await _blobCacheService.DeleteItem(new PublicationCacheKey(publication.Slug));
```

Comment:
I've made the change so now we don't delete the cache file here, but instead delete it in the `PublishStagedReleaseContent` method as you suggested.

We also delete the cache file alongside the deletion of other cache files in PublishReleaseContentFunction#PublishReleaseContent`. I think this is the correct place as in addition to the bug in the above comment, I think it would also be possible for someone to generate a new publication cache file before `_releaseService.SetPublishedDates` which I believe would lead to an issue.